osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs

```diff
@@ -2,210 +2,177 @@
-                    }
+                    if (room?.Host?.Equals(localUser) == true && room.Countdown == null)
+                        setGreen();
                     else
-                    {
-                        button.Text = $"Waiting for host... {countText}";
-                        updateButtonColour(false);
-                    }
+                        setYellow();
 
                     break;
             }
 
-            bool enableButton =
-                Room?.State == MultiplayerRoomState.Open
-                && CurrentPlaylistItem.Value?.ID == Room.Settings.PlaylistItemId
-                && !Room.Playlist.Single(i => i.ID == Room.Settings.PlaylistItemId).Expired
-                && !operationInProgress.Value;
-
-            // When the local user is the host and spectating the match, the "start match" state should be enabled if any users are ready.
-            if (localUser?.State == MultiplayerUserState.Spectating)
-                enableButton &= Room?.Host?.Equals(localUser) == true && newCountReady > 0;
-
-            button.Enabled.Value = enableButton;
-
-            if (newCountReady == countReady)
-                return;
-
-            readySampleDelegate?.Cancel();
-            readySampleDelegate = Schedule(() =>
+            void setYellow()
             {
-                if (newCountReady > countReady)
-                {
-                    if (newCountReady == newCountTotal)
-                        sampleReadyAll?.Play();
-                    else
-                        sampleReady?.Play();
-                }
-                else if (newCountReady < countReady)
-                {
-                    sampleUnready?.Play();
-                }
+                BackgroundColour = colours.YellowDark;
+                Triangles.ColourDark = colours.YellowDark;
+                Triangles.ColourLight = colours.Yellow;
+            }
 
-                countReady = newCountReady;
-            });
+            void setGreen()
+            {
+                BackgroundColour = colours.Green;
+                Triangles.ColourDark = colours.Green;
+                Triangles.ColourLight = colours.GreenLight;
+            }
         }
 
-        private void updateButtonColour(bool green)
+        protected override void Dispose(bool isDisposing)
         {
-            if (green)
-            {
-                button.BackgroundColour = colours.Green;
-                button.Triangles.ColourDark = colours.Green;
-                button.Triangles.ColourLight = colours.GreenLight;
-            }
-            else
-            {
-                button.BackgroundColour = colours.YellowDark;
-                button.Triangles.ColourDark = colours.YellowDark;
-                button.Triangles.ColourLight = colours.Yellow;
-            }
+            base.Dispose(isDisposing);
+
+            if (multiplayerClient != null)
+                multiplayerClient.RoomUpdated -= onRoomUpdated;
```

Comment:
This is incorrect after the `AddDelayed`

Would propose:

```diff
diff --git a/osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs b/osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs
index 6ff717d5c3..746e4257f1 100644
--- a/osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs
+++ b/osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs
@@ -31,7 +31,7 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            multiplayerClient.RoomUpdated += () => Scheduler.AddOnce(onRoomUpdated);
+            multiplayerClient.RoomUpdated += onRoomUpdated;
             onRoomUpdated();
         }
 
@@ -39,7 +39,7 @@ protected override void LoadComplete()
         private DateTimeOffset countdownReceivedTime;
         private ScheduledDelegate countdownUpdateDelegate;
 
-        private void onRoomUpdated()
+        private void onRoomUpdated() => Scheduler.AddOnce(() =>
         {
             if (countdown == null && room?.Countdown != null)
                 countdownReceivedTime = DateTimeOffset.Now;
@@ -56,7 +56,7 @@ private void onRoomUpdated()
 
             updateButtonText();
             updateButtonColour();
-        }
+        });
 
         private void updateButtonText()
         {

```

Might delay the initial application a frame but I don't think that's a huge concern at this point.

Reply:
Uh... Whoops. It should actually be fine since the scheduler is run once after `LoadComplete` (today I learned).